src/runtime/panic.go

```diff
@@ -62,3 +62,10 @@ func sliceBoundsCheckMake(length, capacity uint) {
 		runtimePanic("slice size out of range")
 	}
 }
+
+// Check for bounds in *ssa.MakeSlice. Supports 64-bit indexes.
+func sliceBoundsCheckMake64(length, capacity uint64) {
```

Comment:
a related bug:  sliceBoundsCheckMake params are uint, but it's not guaranteed that uint has the same size as uintptr

Reply:
You are right. It looks like I missed this function in ee7c276493547f1e8ea0ded3f91310ed38d0e1ac (but the bug is older than that).

Reply:
On second thought, no. The length and capacity parameters are _not_ the same thing as what goes into the slice but what is supplied by the user. So it has to handle anything that can go into `make([]T, ...)`, including 64-bit pointers on 32-bit platforms for example.

Reply:
No, my point is, imagine uintptr is 16 bit, int/uint is 32 (Arduino?), len/cap are 16 bit. `biggestInt` will be `uintptr`, but `biggestIntWidth` will be 4. The check `sliceLen.Type().IntTypeWidth() < biggestIntWidth` does fire (2<4), and you "extend" sliceLen to `biggestInt` which is still 16 bit.

And then you call  `sliceBoundsCheckMake` with 16-bit sliceLen/sliceCap, but it expects 32-bit parameters.

Reply:
`c.uintptrType.IntTypeWidth()` is 16 (number of bits) on an Arduino, I just checked.
The `IntTypeWidth` method returns the number of bits the method takes up. The name is a bit confusing, but it does call a method called [`getBitWidth`](https://llvm.org/doxygen/ValueTracking_8cpp.html#a2cb59ea8f9e8543986d40c48acfd24a3) which is more clear in what it returns.